src/Classes/ConfigTab.lua

```diff
@@ -185,6 +185,170 @@ local ConfigTabClass = newClass("ConfigTab", "UndoHandler", "ControlHost", "Cont
+										elseif varExtra[1] == "slider" then
+											control.varControlList[2][i*(#varData.extraTypes)+j-1].val = input[index1] / 100
+											control.varControlList[2][(i+1)*(#varData.extraTypes)+j-1].val = input[index1] / 100
+										end
+									end
+								else
+									break
+								end
+							end
+						end
+					end
+					local firstNone = false
+					for i, varControl in ipairs(control.varControlList[1]) do
+						varControl.selIndex = 1
+						varControl.list = { { val = "NONE", label = "None" } }
+						if i == 1 and not varData.showAll then
+							t_insert(varControl.list, { val = "ALL", label = "All Options" })
+						end
+						varControl.shown = true
+						for j, element in ipairs(varData.list) do
+							if not excludeValue[element.val] or element.val == newValues[i] then
+								t_insert(varControl.list, element)
+							end
+							if element.val == newValues[i] then
+								varControl.selIndex = #varControl.list
+							end
+						end
+						if not varData.showAll then
+							if i == 1 and newValues[i] == "ALL" then
+								varControl.selIndex = 2
+								for j, varControl2 in ipairs(control.varControlList[1]) do
+									if j ~= i then
+										varControl2.shown = false
+									end
+								end
+								break
+							elseif newValues[i] == "NONE" then
+								if firstNone then
+									varControl.shown = false
+								else
+									firstNone = true
+								end
+							end
+						end
+					end
+					for i, varControl in ipairs(control.varControlList[1]) do
+						if i ~= #control.varControlList[1] and varControl:IsShown() then
+							if varData.extraTypes and control.varControlList[2][i * (#varData.extraTypes)]:IsShown() then
+								control.varControlList[1][i+1].y = varControl.y + control.varControlList[2][i * (#varData.extraTypes)].y + m_max(control.varControlList[2][i * (#varData.extraTypes)].height, 18) + 4
+							else
+								control.varControlList[1][i+1].y = varControl.y + m_max(varControl.height, 18) + 4
+							end
+						else
+							break
+						end
+					end
+				end
+				local extraHeight = 20
+				if varData.listFunc then
+					varData.list = varData.listFunc(self.build)
+				end
+				for i=1,(varData.maxElements or #varData.list) do
+					local dropDownControl = new("DropDownControl", {"TOPLEFT",control,"TOPLEFT"}, -225, extraHeight, 343, 18, { { val = "NONE", label = "None" } }, function(index, value)
+						self.configSets[self.activeConfigSetId].input[varData.var.."_"..i] = value.val
+						control.newLists()
+						self:AddUndoState()
+						self:BuildModList()
+						self.build.buildFlag = true
+					end)
+					dropDownControl.shown = varData.showAll or i == 1
+					dropDownControl.newLists = control.newLists
+					self.varControls[varData.var.."_"..i] = dropDownControl
+					t_insert(control.varControlList[1], dropDownControl)
+					local extraTypesExtraHeight = 0
+					for j, varExtra in ipairs(varData.extraTypes or {}) do
+						if varExtra[1] == "integer" then
```

Comment:
I think we use "count" everywhere else so using it here as well would make things more consistent.

Reply:
Integer is a type, yeah I can swap it to count, most things should use count instead anyway and fixes the other issue with negative numbers